File: src/tmp/Java/src/com/dbc/algorithm/Stack/EvaluateReversePolishNotation.java
package com.dbc.algorithm.Stack;

import java.util.Stack;

public class EvaluateReversePolishNotation {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for (String token : tokens) {
            if (!token.equals("+") && !token.equals("-") && !token.equals("*") && !token.equals("/")) {
                stack.push(Integer.parseInt(token));
            } else {
                int a = stack.pop();
                int b = stack.pop();
                if (token.equals("+")) {
                    stack.push(a + b);
                } else if (token.equals("-")) {
                    stack.push(b - a);
                } else if (token.equals("*")) {
                    stack.push(a * b);
                } else {
                    stack.push(b / a);
                }
            }
        }
        return stack.pop();
    }
}

File: src/tmp/Python/stack/evaluate_reverse_polish_notation.py
from typing import List

def evalRPN(tokens: List[str]) -> int:
    stack = list()
    for token in tokens:
        if token not in "+-*/":
            stack.append(int(token))
        else:
            a = stack.pop()
            b = stack.pop()
            if token == '+':
                stack.append(a + b)
            elif token == '-':
                stack.append(b - a)
            elif token == '*':
                stack.append(a * b)
            else:
                stack.append(int(b / a))
    
    return stack[-1]

if __name__ == "__main__":
    print(evalRPN(["10", "6", "9", "3", "+", "-11", "*", "/", "*", "17", "+", "5", "+"]))